src/reducers/entities/users.ts

```diff
@@ -341,7 +366,11 @@ function statuses(state: RelationOneToOne<UserProfile, string> = {}, action: Gen
     }
     case UserTypes.LOGOUT_SUCCESS:
         return {};
-
+    case UserTypes.PROFILE_NO_LONGER_VISIBLE: {
+        const newState = {...state};
+        delete newState[action.data.user_id];
```

Comment:
you should check that the state actually contains the userId before deleting, just to avoid returning a newState if nothing has changed.